cmd/restic/cmd_prune.go

```diff
@@ -362,10 +364,24 @@ func prune(opts PruneOptions, gopts GlobalOptions, repo restic.Repository, usedB
 		return err
 	}
 
+	// packs that are referenced in the index but are not needed can be ignored
+	ignorePacks := restic.NewIDSet()
+	for id, p := range indexPack {
+		if p.usedBlobs == 0 && p.duplicateBlobs == 0 {
+			ignorePacks.Insert(id)
+			stats.blobs.remove += p.unusedBlobs
+			stats.size.remove += p.unusedSize
+			delete(indexPack, id)
+		}
+	}
+
 	if len(indexPack) != 0 {
-		Warnf("The index references pack files which are missing from the repository: %v\n", indexPack)
+		Warnf("The index references needed pack files which are missing from the repository: %v\n", indexPack)
 		return errorPacksMissing
 	}
+	if len(ignorePacks) != 0 {
+		Verbosef("more pack files referenced in the index than present in the repository\n")
```

Comment:
What is a user supposed to do when reading that warning? Maybe tell the user that prune will repair the damage as these packs are not necessary.